mlos_core/mlos_core/optimizers/flaml_optimizer.py

```diff
@@ -101,54 +102,47 @@ def __init__(
 
     def _register(
         self,
-        *,
-        configs: pd.DataFrame,
-        scores: pd.DataFrame,
-        context: Optional[pd.DataFrame] = None,
-        metadata: Optional[pd.DataFrame] = None,
+        observations: Optional[Union[Observation, Observations]] = None,
     ) -> None:
         """
-        Registers the given configs and scores.
+        Registers the given config and scores.
 
         Parameters
         ----------
-        configs : pd.DataFrame
-            Dataframe of configs / parameters. The columns are parameter names and
-            the rows are the configs.
-
-        scores : pd.DataFrame
-            Scores from running the configs. The index is the same as the index of the configs.
-
-        context : None
-            Not Yet Implemented.
-
-        metadata : None
-            Not Yet Implemented.
+        observation : Observation | Observations
+            The observations to register.
         """
-        if context is not None:
-            warn(f"Not Implemented: Ignoring context {list(context.columns)}", UserWarning)
-        if metadata is not None:
-            warn(f"Not Implemented: Ignoring metadata {list(metadata.columns)}", UserWarning)
-
-        for (_, config), (_, score) in zip(configs.astype("O").iterrows(), scores.iterrows()):
-            # Remove None values for inactive config parameters
-            config_dict = drop_nulls(config.to_dict())
-            cs_config: ConfigSpace.Configuration = ConfigSpace.Configuration(
-                self.optimizer_parameter_space,
-                values=config_dict,
+        assert isinstance(
+            observations, Observation
+        ), "Internal implementation does not support Observations."
```

Comment:
Why not?
At the very least it could convert an Observations object with multiple Observation instances to a singleton.

Reply:
I'd rather not mix too many changes at once here.

If the previous implementation handled registering multiple scores at once, we should retain that.

If you want to refactor it later to restrict it to only handle registering one at a time or doing that via a loop internally, we should make that change separately.

This one should only be about changing the signatures to take a new encapsulated data type.